### PR TITLE
GPII-2077:  Remove local copies of transforms now in infusion...

### PR DESCRIPTION
**Pull request by the-t-in-rtf**
See [GPII-2077](https://issues.gpii.net/browse/GPII-2077) for details.

### CODE REVIEW

**Comment by the-t-in-rtf** [comment]
Just to comment on priorities: this should be reviewed once [FLUID-5971](https://github.com/fluid-project/infusion/pull/761) is merged.

**Comment by the-t-in-rtf** [comment]
FYI, the tests fail at the moment because they explicitly confirm that model deletions are relayed, which they do not seem to be.

**Comment by amb26** [comment]
Could we fix up the tests to lighten the dependency on testem? It's no longer possible to start them without it

**Comment by amb26** [comment]
Sorry, please ignore previous comment

**Comment by amb26** [comment]
Sorry, please unignore the comment :)

**Comment by the-t-in-rtf** [comment]
Hi, @amb26.  Just to document our chat and the results here:

1. We discussed the hard requirement on testem.
2. I updated the package to move the instrumentation and reporting out of the testem component.

This left us with the requirement to instrument the code before opening the test fixtures.  I tried adding this as a `postinstall` step, but typically enough, even though this is only an `npm run` call, it only works within the package itself.  The failure blocks installing gpii-binder as a dependency.  I have removed the `postinstall` step and updated the README to indicate that `npm run instrument` must be run before you can use the tests outside of testem.  I hope this will be enough.

**Comment by amb26** [comment]
Not sure I get it - isn't the postinstall step only required in order to build instrumentation for test fixtures? Why would the absence of this block installing gpii-binder as a dependency? We should try hard to make sure that all projects conform to standard expectations that "npm install" will bring them into a condition where tests can be run immediately

**Comment by amb26** [comment]
You may like to consult a strategy applied by @cindyli in order to get a postinstall "browserify" step to run, following the pattern you suggested for the in-browser DataSources: https://github.com/cindyli/universal/blob/GPII-1274/scripts/browserifyTestDependency.js#L34

**Comment by amb26** [comment]
Also, your "instrumentation by default" strategy makes the resulting test cases impossible to debug - please consult attached screenshot:
instrument

**Comment by amb26** [comment]
OK, now I squint at it a bit more, "impossible" seems a bit strong ...

**Comment by the-t-in-rtf** [comment]
In other packages, I can cheat by mounting different content at the same path.  I can see the "instrumented by default" strategy is problematic.  I plan to make a parallel test file and move the instrumentation to "pretest", which is more appropriate.

**Comment by the-t-in-rtf** [comment]
I was able to come up with a strategy that avoids creating separate files.  Basically, we can remap /src to /instrumented within the testem configuration using their ["routes"](https://github.com/testem/testem/blob/241b1e42e96cd7e4d656efda6475e96e1f7978ae/examples/routes/testem.json) mechanism.  Thus the files can be opened directly and will also collect coverage data during the test runs.

**Comment by the-t-in-rtf** [comment]
The new approach is in place, which will hopefully make it easier to look into the deletion relaying concerns (thanks for your help there).

**Comment by the-t-in-rtf** [comment]
I collapsed the two previous tests into a single "apply a change once, check the results in two places" pattern.  @amb26 is working on the deletion relay fix.

With the previous approach there was some concern that applying the same change twice to the same component in subsequent test runs was causing problems.

**Comment by the-t-in-rtf** [comment]
Although I can verify the tests are working, I am seeing spurious test failures with browser disconnects when running testem.  I could use help investigating.  My notes to date:

1. Happens in yarn and npm, but fails more quickly in npm.
2. Fails whether or not we use the all-tests page.
3. Fails whether or not we use the “standard” caseHolder.
4. All tests fail/hang individually when they are the only thing run, at least using all-tests.

I could use help in confirming whether this is unique to my setup (browser versions, etc.) and continuing the investigation.  My current running theories are that either there is an outright bug in newer versions of testem, or that their QUnit harness has finally moved past the point where it can work with our version of QUnit.

**Comment by the-t-in-rtf** [comment]
I looked through the commit history to see what version(s) we had before.  I can avoid the disconnects by downgrading to testem 1.9.1, but then the tests hang at the very end after the last line of output.

**Comment by the-t-in-rtf** [comment]
[This (closed) issue](https://github.com/testem/testem/issues/777) in testem's issue tracking seems relevant, at least in giving ideas for tracking it down further.

**Comment by the-t-in-rtf** [comment]
@amb26, as discussed, we need to confirm whether the intermittent testem failures occur on vagrant and agree (for now) whether having the tests pass in vagrant is adequate to merge this, or whether more research is needed.

**Comment by the-t-in-rtf** [comment]
I figured out the cause of the "browser undefined disconnected errors".  See [GPII-2264](https://issues.gpii.net/browse/GPII-2264) for details.  @amb26, this should be ready for review.

File: package.json [hunk 1]
```diff
@@ -5,22 +5,24 @@
   "license": "BSD-3-Clause",
   "repository": "https://github.com/GPII/gpii-binder",
   "scripts": {
-    "pretest": "node node_modules/rimraf/bin.js coverage instrumented",
-    "test": "node node_modules/testem/testem.js ci --skip PhantomJS,Chromium,Safari"
+    "instrument": "node node_modules/istanbul/lib/cli.js instrument --no-compact --output instrumented src",
+    "pretest": "node node_modules/rimraf/bin.js coverage instrumented && npm run instrument",
+    "test": "node node_modules/testem/testem.js ci --skip PhantomJS,Chromium,Safari",
```
Comment (amb26) [inline package.json:10]: Why do we skip Chromium and Safari here?

Comment (the-t-in-rtf) [reply package.json:10]: Safari has had intermittent errors loading testem itself, once we have OS X available as a QI target I will revisit in more depth.  Chromium is probably safe now, but needs to be confirmed in QI.

**Pull request merged**